File: chapter_5/typescript/check_voter.ts
interface HashTable<T> {
    [key: string]: T;
}
const voted: HashTable<boolean> = {};

const checkVoter = (name: string) => {
    if (voted[name]) {
        console.log('Kick them out!!');
    } else {
        voted[name] = true;
        console.log('Let them vote!')
    }
}

checkVoter("tom"); // let them vote!
checkVoter("mike"); // let them vote!
checkVoter("mike"); // kick them out!

File: chapter_5/golang/check_voter.go
package main

import "fmt"

var voted map[string]bool

func check_voter(name string) {
	if voted[name] {
		fmt.Println("Kick them out!")
	} else {
		voted[name] = true
		fmt.Println("Let them vote!")
	}
}

func main() {
	voted = make(map[string]bool)
	check_voter("tom")
	check_voter("mike")
	check_voter("mike")
}
